packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/FlutterNextStepUtility.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_auth_cognito
+
+import com.amplifyframework.auth.AuthCodeDeliveryDetails
+import com.google.gson.Gson
+
+
+fun setNextStep(resultKey: String, resultValue: String, codeDeliveryDetails: AuthCodeDeliveryDetails?, additionalInfo: Map<String, String>?): Map<String, Any> {
+   val result =  mutableMapOf<String, Any>(resultKey to resultValue)
+   val codeDeliveryDetailsResult = mutableMapOf<String, String?>()
+
+   if (codeDeliveryDetails is AuthCodeDeliveryDetails) {
```

Comment:
Are these changes needed for null safety? Anything passed over the method channel could be null. Do any of the changes in kotlin/swift impact nullability on the dart side?

FYI - Some of these refactors are going to conflict with changes I have in [an open PR](https://github.com/aws-amplify/amplify-flutter/pull/601/files#diff-96450b5b67f95088a777b70333ae932561fcf1cb8940438a6c88fc7771b53469R21). Not a huge deal. We can discuss how we want to reconcile it before either PR is merged potentially. I would suggest we limit how much refactoring is done in the null safety branches though (or any change of this scope). There could be a lot of drift between the null safety branches and master.

Reply:
After looking again, I don't think these will actually conflict since you didn't consume this utility in `FlutterUpdateUserAttributeResult.kt` or `FlutterUpdateUserAttributesResult.kt`. We will have two similar utils but I guess they could be combined later.

Should `FlutterUpdateUserAttributeResult.kt` and `FlutterUpdateUserAttributesResult.kt` be updated? Was there an issue with how the next step was being set or were you just trying to consolidate duplicated logic?

Reply:
@haverchuck could you provide your insights here.  I think Dustin might have added this as part of his code updates so I think he should have the final say on the best next step.  From what I understand/see I'm guessing these should be consolidated.

Reply:
We were just trying to consolidate code with this file - it was requested as part of code review. 
I think it would make sense to update the FlutterUpdateUserAttributeResult class to use this same utility.